### PR TITLE
[FEATURE] Delete Category

### PR DESCRIPTION
**Pull request by MatejMecka**
Please prefix your pull request with one of the following: **[FEATURE]** **[FIX]** **[IMPROVEMENT]**.

**In raising this pull request, I confirm the following (please check boxes):**

- [X] I have read and understood the [contributors guide](https://github.com/CCExtractor/sample-platform/blob/master/.github/CONTRIBUTING.md).
- [X] I have checked that another pull request for this purpose does not exist.
- [X] I have considered, and confirmed that this submission will be valuable to others.
- [X] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [X] I give this submission freely, and claim no ownership to its content.

**My familiarity with the project is as follows (check one):**

- [ ] I have never used the project.
- [ ] I have used the project briefly.
- [ ] I have used the project extensively, but have not contributed previously.
- [X] I am an active contributor to the project.

---

Delete Category

### CODE REVIEW

File: tests/test_regression/TestControllers.py [hunk 1]
```diff
@@ -30,3 +30,48 @@ def test_regression_test_status_toggle(self):
...
+
+    def category_delete(self):
+        """
+        Check it will delete the test
+        :return:
+        """
+        # Create Valid Entry
+        from mod_regression.models import Category, RegressionTest
+
+        category = Category('Some Random Category', 'Todo')
```
Comment (satyammittal) [inline tests/test_regression/TestControllers.py]: You don't need to add categories. They have been added already in base.py. This category will not have id 1.

File: tests/test_regression/TestControllers.py [hunk 2]
```diff
@@ -58,10 +58,6 @@ def category_delete(self):
         :return:
         """
         # Create Valid Entry
-        from mod_regression.models import Category, RegressionTest
-
-        category = Category('Some Random Category', 'Todo')
-
         g.db.add(category)
```
Comment (satyammittal) [inline tests/test_regression/TestControllers.py]: remove this :)

File: tests/test_regression/TestControllers.py [hunk 3]
```diff
@@ -58,10 +58,6 @@ def category_delete(self):
...
         g.db.commit()
```
Comment (satyammittal) [inline tests/test_regression/TestControllers.py]: remove this :)

File: mod_regression/controllers.py [hunk 4]
```diff
@@ -121,9 +119,24 @@ def test_add():
...
-    # Delete a regression test category
-    pass
+    """
+    Delete the category
+    :param category_id: The ID of the Category
+    :type int
+    :return: Redirect
+    """
+
+    category = Category.query.filter(Category.id == category_id).first()
```
Comment (canihavesomecoffee) [inline mod_regression/controllers.py]: I'm missing a confirmation page to delete the category.

**Comment by MatejMecka** [comment]
Closing, because i don't know what i did anymore to my fork. Will open another pr

**Pull request closed without merging**